stl/inc/xstring

```diff
@@ -4815,26 +4831,13 @@ private:
         _Traits::assign(_Mypair._Myval2._Myptr()[_Mypair._Myval2._Mysize = _New_size], _Elem());
     }
 
-    _CONSTEXPR20 void _Tidy_init() noexcept {
-        // initialize basic_string data members
-        auto& _My_data   = _Mypair._Myval2;
-        _My_data._Mysize = 0;
-        _My_data._Myres  = _Small_string_capacity;
-        _My_data._Activate_SSO_buffer();
-
-        // the _Traits::assign is last so the codegen doesn't think the char write can alias this
-        _Traits::assign(_My_data._Bx._Buf[0], _Elem());
-    }
-
-    _CONSTEXPR20 void _Tidy_deallocate() noexcept { // initialize buffer, deallocating any storage
+    _CONSTEXPR20 void _Tidy_deallocate() noexcept { // switch to small mode, deallocating any storage
         auto& _My_data = _Mypair._Myval2;
         _My_data._Orphan_all();
         if (_My_data._Large_mode_engaged()) {
             _ASAN_STRING_REMOVE(*this);
-            const pointer _Ptr = _My_data._Bx._Ptr;
+            const pointer _Ptr = _My_data._Release_ptr_and_activate_buffer();
             auto& _Al          = _Getal();
-            _Destroy_in_place(_My_data._Bx._Ptr);
-            _My_data._Activate_SSO_buffer();
             _Deallocate_for_capacity(_Al, _Ptr, _My_data._Myres);
         }
 

```

Comment:
Slightly irrelevant; it also seems that the block in this function is actually useless (`_Tidy_deallocate` seems always followed by methods that just rewrite these values).
```
        _My_data._Mysize = 0;
        _My_data._Myres  = _Small_string_capacity;
        // the _Traits::assign is last so the codegen doesn't think the char write can alias this
        _Traits::assign(_My_data._Bx._Buf[0], _Elem());
```